tests/system/small/test_dataframe.py

```diff
@@ -2465,6 +2465,24 @@ def test_df_describe(scalars_dfs):
     ).all()
 
 
+def test_df_transpose():
```

Comment:
If it's a cheap test, we could add a check that we raise TypeError for mixing strs and floats.

Reply:
added